File: Maximum_Length_of_a_Concatenated_String_with_Unique_Characters.cpp
class Solution {
public:
    int maxLength(vector<string>& arr) {
        vector<string> uniqueComb = {""};
        int maxLengthResult = 0;

        for (int i = 0; i < arr.size(); i++) {
            for (int j = 0; j < uniqueComb.size(); j++) {
                string newComb = arr[i] + uniqueComb[j];

                if (unordered_set<char>(newComb.begin(), newComb.end()).size() == newComb.length()) {
                    uniqueComb.push_back(newComb);
                    maxLengthResult = max(maxLengthResult, static_cast<int>(newComb.length()));
                }
            }
        }

        return maxLengthResult;        
    }
};

File: Maximum_Length_of_a_Concatenated_String_with_Unique_Characters.py
class Solution:
    def maxLength(self, arr: List[str]) -> int:
        uniq_comb = [""]
        max_length = 0

        for i in range(len(arr)):
            for j in range(len(uniq_comb)):
                new_comb = arr[i] + uniq_comb[j]

                if len(set(new_comb)) == len(new_comb):
                    uniq_comb.append(new_comb)
                    max_length = max(max_length, len(new_comb))
        
        return max_length
